### ISSUE TITLE
Add an RNG service and avoid using `TRandom`

### ISSUE THREAD

**Issue by c-dilks** [eic/EICrecon#539]
**Is your feature request related to a problem? Please describe.**
Problem: In the algorithm `PhotoMultiplierHitDigi`, if the seed for the `TRandomMixMax` object is set to 0 (use "random" seed, see [documentation](https://root.cern.ch/doc/master/classTRandom.html#a2007ae15ce828266aecb783ed3410e8b)), then the randomly generated number can sometimes remain the same for several generations, then start actually randomizing later; this is a possible symptom of thread unsafety. This issue does not appear to happen if the seed is non-zero (a fixed seed, useful for reproducibility, but less random than seed 0).

`TRandom` is used in the following algorithms and factories:
```
src/algorithms/digi/PhotoMultiplierHitDigi.h
src/algorithms/digi/SiliconTrackerDigi.h
src/algorithms/reco/MC2SmearedParticle.h
src/global/digi/SiliconTrackerDigi_factory.h
src/global/tracking/TrackerHitReconstruction_factory.h
```
and some tracking efficiency benchmarks.

**Describe the solution you'd like**
Ideally, we should implement a thread-safe random number generation service.

**Describe alternatives you've considered**
Use seed != 0, which appears to work okay, but is not really ideal for the long term, especially if we want more randomness.

**Comment by DraTeots** [comment]
BTW, there is [random generator service](https://github.com/eic/EICrecon/tree/main/src/services/randomgenerator) but it is barely usable in the current implementation.

**Comment by c-dilks** [comment]
Nice, let's make it a proper service (e.g., `src/RandGen_driver.cpp` should not live in `src`) and start using it! Unfortunately I don't have time to help do this, but this seems like a very good entry-level issue for a new person to get involved with.

**Comment by wdconinc** [comment]
> this seems like a very good entry-level issue for a new person to get involved with.

Not quite. Getting random numbers isn't hard. It's getting the same random numbers in the same sequence to the same algorithms, even when algorithms may run out of sequence the next time you run it... And this must all work in multithreaded running.

**Comment by c-dilks** [comment]
> BTW, there is [random generator service](https://github.com/eic/EICrecon/tree/main/src/services/randomgenerator) but it is barely usable in the current implementation.

Removed by https://github.com/eic/EICrecon/pull/640.
